tests/service/views/test_dataset_views.py

```diff
@@ -807,3 +807,195 @@ def test_import_dataset_job_enqueue(
+    }
+
+    user = cache.ensure_user({'user_id': 'user'})
+
+    project_meta = {
+        'project_id': uuid.uuid4().hex,
+        'name': Path(project).name,
+        'fullname': 'full project name',
+        'email': 'my@email.com',
+        'owner': 'me',
+        'token': 'awesome token',
+        'git_url': 'git@gitlab.com'
+    }
+
+    project_obj = cache.make_project(user, project_meta)
+
+    dest = project_obj.abs_path
+    os.makedirs(dest.parent, exist_ok=True)
+    if not (project / dest).exists():
+        shutil.copytree(project, dest)
+
+    response = client.post(
+        '/datasets.add',
+        data=json.dumps({
+            'project_id': project_meta['project_id'],
+            'dataset_name': uuid.uuid4().hex,
+            'create_dataset': True,
+            'files': [{
+                'file_url': url
+            }]
+        }),
+        headers=headers,
+    )
+
+    assert_rpc_response(response)
+    assert {'files', 'dataset_name',
+            'project_id'} == set(response.json['result'])
+    job_id = response.json['result']['files'][0]['job_id']
+
+    user_job = cache.get_job(user, job_id)
+    assert job_id == user_job.job_id
+
+    response = client.get('/jobs', headers=headers)
+    assert_rpc_response(response)
+    assert response.json['result']['jobs']
+
+    assert user_job.job_id in [
+        job['job_id'] for job in response.json['result']['jobs']
+    ]
+
+
+@pytest.mark.integration
+@pytest.mark.service
+@flaky(max_runs=30, min_passes=1)
+def test_dataset_add_multiple_remote(svc_client_cache, project, mock_redis):
+    """Test dataset add multiple remote files."""
+    url_gist = 'https://gist.github.com/jsam/d957f306ed0fe4ff018e902df6a1c8e3'
+    url_dbox = 'https://www.dropbox.com/s/qcpts6fc81x6j4f/addme?dl=0'
+
+    client, cache = svc_client_cache
+
+    headers = {
+        'Content-Type': 'application/json',
+        'Renku-User-Id': 'user',
+        'Renku-User-FullName': 'full name',
+        'Renku-User-Email': 'renku@sdsc.ethz.ch',
+    }
+
+    user = cache.ensure_user({'user_id': 'user'})
+
+    project_meta = {
+        'project_id': uuid.uuid4().hex,
+        'name': Path(project).name,
+        'fullname': 'full project name',
+        'email': 'my@email.com',
+        'owner': 'me',
+        'token': 'awesome token',
+        'git_url': 'git@gitlab.com'
+    }
+
+    project_obj = cache.make_project(user, project_meta)
+
+    dest = project_obj.abs_path
+    os.makedirs(dest.parent, exist_ok=True)
+    if not (project / dest).exists():
+        shutil.copytree(project, dest)
+
+    response = client.post(
+        '/datasets.add',
+        data=json.dumps({
+            'project_id': project_meta['project_id'],
+            'dataset_name': uuid.uuid4().hex,
+            'create_dataset': True,
+            'files': [{
+                'file_url': url_gist,
+            }, {
+                'file_url': url_dbox,
+            }]
+        }),
+        headers=headers,
+    )
```

Comment:
This code has a lot of redundancy with the above function. Please consider to factor it to a function.

Reply:
done.